### PR TITLE
Enhancements to Freshdesk Stream Handling

### PR DESCRIPTION
**Pull request by Prashantvik**
_Enhancements to Freshdesk Stream Handling_

**Change 1:**
- **Function Override:** `response_error_message` in `FreshdeskStream` class.
- **Parent Class:** `RESTStream` from Singer SDK.
- **Resolution:** Added error details to enable better error tracing. Each error message is now printed with its associated fields/types.

**Change 2:**
- **Property Override:** `is_sorted` in `TicketsAbridgedStream`.
- **Parent Class:** `PagedFreshdeskStream` from Singer SDK.
- **Resolution:** Implemented sorting strategy by `updated_at` in ascending order for tickets_abridged streams. This adjustment ensures data processed prior to a break won't be reprocessed, considering the default API design orders data in descending order.

**Change 3:**
- **Function Return Condition:** `has_more` function.
- **Resolution:** Addressed the issue where the stream couldn't identify when to stop due to the API's default pagination design. Added the condition `self.current_value < 300` to handle scenarios where pagination count exceeds 300 pages.

**Change 4:**
- **Parameter Addition:** Function `get_url_params` in class `TicketsAbridgedStream`.
- **Resolution:** Added `order_by` and `order_type` parameters to facilitate ascending sorting of streams. Implemented via the `is_sorted` function introduced earlier.

**Change 5:**
- **Schema change:** for tickets schema - file tickets.json
- **Resolution:** Incoming stream from fresdesk has datatype of number for key association_type. Please refer : https://developers.freshdesk.com/api/

This PR enhances Freshdesk stream handling, ensuring better error tracing, proper sorting, and pagination handling.

### CODE REVIEW

File: tap_freshdesk/client.py [hunk 1]
```diff
@@ -183,6 +186,48 @@ def _wait_for(exception) -> int:
...
+            try:
+                error_data = response.json()
+                errors = error_data.get('errors')
+                index = 0
+                for error in errors:
+                    message = error.get('message', 'Unknown')
+                    field = error.get('field', 'Unknown')
+                    error_details.append(f"Error {index + 1}: Message - {message}, Field - {field}")
+                    index += 1
+            except json.JSONDecodeError:
```
Comment (acarter24) [inline tap_freshdesk/client.py]: I think this needs to be `requests.exceptions.JSONDecodeError` not `json.JSONDecodeError`.

```
requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

File: tap_freshdesk/client.py [hunk 2]
```diff
@@ -183,6 +186,48 @@ def _wait_for(exception) -> int:
...
+        )
+
+        error_details = []
+        if response.status_code >= 400:
+            print(f"Error Response: {response.status_code} {response.reason}")
+            try:
+                error_data = response.json()
+                errors = error_data.get('errors')
+                index = 0
+                for error in errors:
```
Comment (acarter24) [inline tap_freshdesk/client.py]: Please use `enumerate(errors)` here, no need to track index as a separate variable.

`for index, error in enumerate(errors):`

**Review by acarter24 — CHANGES_REQUESTED** [review]
Just a couple comments. I suspect we are using a lower level licence than you so we have to respect the timeout limits of the API, hence the json exception being flagged.

File: tap_freshdesk/client.py (hunk 1 above)
Comment (Prashantvik) [reply tap_freshdesk/client.py]: Got it, this has been implemented using `requests.exceptions.JSONDecodeError`

File: tap_freshdesk/client.py (hunk 2 above)
Comment (Prashantvik) [reply tap_freshdesk/client.py]: Changes implemented as suggested.

**Pull request merged**